diff -uNr python-paradox-alarm-interface-3.1.0.orig/setup.py python-paradox-alarm-interface-3.1.0.new/setup.py
--- python-paradox-alarm-interface-3.1.0.orig/setup.py	2022-12-25 23:16:15.527957916 +0200
+++ python-paradox-alarm-interface-3.1.0.new/setup.py	2022-12-25 23:12:08.000000000 +0200
@@ -25,7 +25,7 @@
     packages=find_packages(exclude=["tests", "tests.*", "config.*", "docs.*"]),
     install_requires=["construct~=2.9.43", "argparse>=1.4.0", "python-slugify>=4.0.1"],
     python_requires=">=3.8",
-    setup_requires=["wheel"] + pytest_runner,
+    setup_requires=pytest_runner,
     tests_require=[
         "pytest",
         "pytest-asyncio>=0.17",
